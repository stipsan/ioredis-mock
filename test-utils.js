// Runs both a regular string suite and a buffer suite
export function runTwinSuite(commandName, cb) {
  const commands = [commandName, `${commandName}Buffer`]

  return commands.forEach(command =>
    command === commandName
      ? cb(command, (a, b) => a === b)
      : cb(command, (a, b) => a.equals(Buffer.from(b)))
  )
}
