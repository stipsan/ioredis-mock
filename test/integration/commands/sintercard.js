import Redis from 'ioredis'

// eslint-disable-next-line import/no-relative-parent-imports
import { runTwinSuite } from '../../../test-utils'

runTwinSuite('sintercard', command => {
  // @TODO: implement this command
  ;(process.env.IS_E2E ? describe : describe.skip)(command, () => {
    const redis = new Redis()

    beforeEach(async () => {
      await Promise.all(
        [
          ['key1', 'a'],
          ['key1', 'b'],
          ['key1', 'c'],
          ['key1', 'd'],
          ['key2', 'c'],
          ['key2', 'd'],
          ['key2', 'e'],
          // eslint-disable-next-line arrow-body-style
        ].map(args => redis.sadd(...args))
      )
    })

    afterAll(() => {
      redis.disconnect()
    })

    test('SINTERCARD 2 key1 key2', async () => {
      const result = await redis[command](2, 'key1', 'key2')
      expect(result).toMatchSnapshot()
    })

    test('SINTERCARD 2 key1 key2 LIMIT 1', async () => {
      const result = await redis[command](2, 'key1', 'key2', 'LIMIT', 1)
      expect(result).toMatchSnapshot()
    })
  })
})
