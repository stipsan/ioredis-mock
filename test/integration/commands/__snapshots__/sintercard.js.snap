// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sintercard SINTERCARD 2 key1 key2 1`] = `2`;

exports[`sintercard SINTERCARD 2 key1 key2 LIMIT 1 1`] = `1`;

exports[`sintercardBuffer SINTERCARD 2 key1 key2 1`] = `2`;

exports[`sintercardBuffer SINTERCARD 2 key1 key2 LIMIT 1 1`] = `1`;
